### PR TITLE
Demo: Demonstrate event traversal through scene graph

### PR DESCRIPTION
**Pull request by darrenjaneczek**
- Added a demo to show how bubble events traverse the scene graph.

- Added a sceneGraph.findDescendents (by type) utility function

[Screencast_20240725_145040.webm](https://github.com/user-attachments/assets/8c886a6c-c448-4b1a-b6ab-5c752763b844)

### CODE REVIEW

File: packages/scenes/src/core/sceneGraph/sceneGraph.ts [hunk 1]
```diff
@@ -219,13 +219,18 @@ export function getDataLayers(sceneObject: SceneObject, localOnly = false): Scen
   return collected;
 }
 
+interface Type<T> extends Function {
+  new (...args: never[]): T;
+}
```
Comment (itsmylife) [inline packages/scenes/src/core/sceneGraph/sceneGraph.ts:224]: 👍

File: packages/scenes/src/core/sceneGraph/sceneGraph.ts [hunk 2]
```diff
@@ -219,13 +219,18 @@ export function getDataLayers(sceneObject: SceneObject, localOnly = false): Scen
...
+
+
```
Comment (itsmylife) [inline packages/scenes/src/core/sceneGraph/sceneGraph.ts:226]:
```suggestion
```

File: packages/scenes/src/core/sceneGraph/sceneGraph.ts (hunk 1 above)
Comment (itsmylife) [reply packages/scenes/src/core/sceneGraph/sceneGraph.ts:224]: `Type` is pretty generic name. Should we use a different name? `AncestorType` maybe?

**Review by itsmylife — APPROVED** [review]
some small nit-picks
the rest 🚀

File: packages/scenes/src/core/sceneGraph/sceneGraph.ts [hunk 3]
```diff
@@ -243,6 +247,18 @@ export function getAncestor<ParentType>(
...
+ */
+export function findDescendents<T extends SceneObject>(scene: SceneObject, descendentType: SceneType<T>) {
+  function isDescendentType(scene: SceneObject): scene is T {
+    return scene instanceof descendentType;
+  }
+
+  const targetScenes = findAllObjects(scene, isDescendentType);
+  return targetScenes.filter(isDescendentType);
+}
+
```
Comment (dprokop) [inline packages/scenes/src/core/sceneGraph/sceneGraph.ts:261]: Test please :)

**Review by dprokop — APPROVED** [review]
Nice!

**Review by dprokop — APPROVED** [review]

**Pull request merged**